libc-top-half/stub-pthreads/spinlock.c

```diff
@@ -0,0 +1,21 @@
+
```

Comment:
Should we call this directory `pthread-stubs`?

And maybe it should live at the top level alongside `dlmalloc` and `emmalloc` (which override/replace libc functions too).  @sunfishcode any preference?